### PR TITLE
fix: upgrade api and support sqllit with raft backend

### LINKED ISSUE #552: [Improvement] upgrade orchestrator addon
- [x] upgrade to 090 API
- [x] support raft mode and share-end mode
   - [x] upgrade apecloud mysql addon
- [x] add role probe and health check
- [ ]  add backup policy template for RAFT mode (to backup sqlite database)

### PR DESCRIPTION
**Pull request by shanshanying**
- fix #552 
in STS, podManagementPolicy and updateStrategy are used to deal with provision and update scenarios resp. 
 the question here is how to provision multiple pod in parallel ( so that they form a valid peers as soon as possible) and update takes effects one by one.

supported two modes to provision an orc cluster, namely
- raft mode
- share-end mode

And add role proble, health check to each mode.
Both mode supports hscale, vscale, but no reconfiguration. (in RAFT mode, we'd better restart the cluster are hscale)

image

## RAFT mode
To create such a cluster with RAFT mode, simply use
```bash
 helm install myorc-raft ./addons/orchestrator-cluster  --set mode=raft
```

## Share-end mode
To create a cluster with SHARE-END mode, 
- step 1. firstly we provisioin a raft-group mysql cluster (Apecloud MySQL) 
with audilog and proxy enabled
```bash
helm install mycluster ./addons/apecloud-mysql-cluster --set auditLogEnabled=true,proxyEnabled=true,mode=raftGroup
```
- step 2. then we create a shareend orc cluster:
```bash
 helm install myorc-raft ./addons/orchestrator-cluster  --set mode=shareend --set serviceReference.metadb.cluster="mycluster"
```

remember, the `mycluster` name refers to the apecloud-mysql cluster we created in step 1.
```

### CODE REVIEW

**Review by xuriwuyun — APPROVED** [review]

File: addons/orchestrator/templates/_helpers.tpl [hunk 1]
```diff
@@ -71,4 +71,118 @@ orchestrator
...
+  - -c
+  - |
+    /scripts/startup.sh
+volumeMounts:
+  - name: configs
+    mountPath: /configs
+  - name: scripts
+    mountPath: /scripts
+  - mountPath: /data
+    name: data
```
Comment (leon-inf) [inline addons/orchestrator/templates/_helpers.tpl:173]: Should define the data volume explicitly.

File: addons/orchestrator-cluster/templates/_helpers.tpl [hunk 2]
```diff
@@ -1,18 +1,20 @@
 {{/*
 Create extra env
 */}}
+# TOPOLOGY and META user name are fixed to orchestrator
+# "ORC_TOPOLOGY_USER": "{{ .Values.secret.TOPOLOGY_USER | default "orchestrator" }}",
+# "ORC_META_USER": "{{ .Values.secret.META_USER | default "orchestrator" }}",
 {{- define "orchestrator-cluster.extra-envs" }}
 {
-"ORC_TOPOLOGY_USER": "{{ .Values.secret.TOPOLOGY_USER | default "orchestrator" }}",
 "ORC_TOPOLOGY_PASSWORD": "{{ .Values.secret.TOPOLOGY_USER | default "orchestrator" }}",
```
[the remark this answers is not in the document]
Comment (shanshanying) [reply addons/orchestrator-cluster/templates/_helpers.tpl]: fixed.

File: addons/orchestrator-cluster/templates/cluster.yaml [hunk 3]
```diff
@@ -4,27 +4,33 @@ metadata:
...
+  terminationPolicy: {{ .Values.extra.terminationPolicy }}
   {{- include "kblib.affinity" . | indent 2 }}
   componentSpecs:
     - name: orchestrator
-      componentDef: orchestrator # ref componentDefinition metadata.name
       {{- include "kblib.componentMonitor" . | indent 6 }}
       replicas: {{ .Values.replicas }}
-      serviceAccountName: {{ include "kblib.serviceAccountName" . }}
       {{- include "kblib.componentResources" . | indent 6 }}
       {{- include "kblib.componentServices" . | indent 6 }}
```
[the remark this answers is not in the document]
Comment (shanshanying) [reply addons/orchestrator-cluster/templates/cluster.yaml]: removed

File: addons/orchestrator/templates/_helpers.tpl [hunk 4]
```diff
@@ -71,4 +71,118 @@ orchestrator
...
+  - name: default
+    roleSelector: primary
+    spec:
+      ports:
+        - name: http
+          port: 80
+          targetPort: http
+        - name: prometheus
+          port: 9125
+          targetPort: prometheus
```
[the remark this answers is not in the document]
Comment (shanshanying) [reply addons/orchestrator/templates/_helpers.tpl]: removed

File: addons/orchestrator/templates/_helpers.tpl (hunk 1 above)
Comment (shanshanying) [reply addons/orchestrator/templates/_helpers.tpl:169]: used info from .Values.

**Review by leon-inf — APPROVED** [review]

**Pull request merged**